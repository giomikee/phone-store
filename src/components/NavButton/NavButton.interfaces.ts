import type { ButtonOwnProps } from '@mui/material';
import type { SIZES } from '../../constants';
import type { ReactNode } from 'react';

export interface Props {
    to: string;
    children: ReactNode;
    startIcon?: ReactNode;
    color?: ButtonOwnProps['color'];
    size?: keyof typeof SIZES;
}