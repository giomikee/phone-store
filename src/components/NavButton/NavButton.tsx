import { NavLink } from 'react-router-dom';
import type { Props } from './NavButton.interfaces';
import { Button } from '@mui/material';

const NavButton = ({ color, children, size, startIcon, to }: Props) => {
    return (
        <NavLink to={to}>
            {
                ({isActive}) => (
                    <Button 
                        color={color}
                        size={size}
                        startIcon={startIcon}
                        variant={ isActive ? 'outlined' : 'text'}
                    >
                        {children}
                    </Button>
                )
            }
        </NavLink>
    )
}

export default NavButton