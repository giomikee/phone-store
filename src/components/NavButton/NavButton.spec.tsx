import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/vitest';
import NavButton from './NavButton';
import { BrowserRouter } from 'react-router-dom';

describe('Given NavButton component', () => {
    const defaultProps = {
        to: '/',
    };

    describe('When the component is rendered', () => {
        it('Then it should match the snapshot', () => {
            const { container } = render(
                <NavButton {...defaultProps}>
                    <span>Home</span>
                </NavButton>, 
                { wrapper: BrowserRouter }
            );
            expect(container).toMatchSnapshot();
        });
    });
})