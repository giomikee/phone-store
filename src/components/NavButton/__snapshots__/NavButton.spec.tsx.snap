// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given NavButton component > When the component is rendered > Then it should match the snapshot 1`] = `
<div>
  <a
    aria-current="page"
    class="active"
    href="/"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-1blvszr-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <span>
        Home
      </span>
    </button>
  </a>
</div>
`;
